pkg/provider/pods.go

```diff
@@ -404,6 +404,19 @@ func (p *Pod) IsRunAsUserID(uid int64) bool {
 	return *p.Pod.Spec.SecurityContext.RunAsUser == uid
 }
 
+func (p *Pod) IsRunAsNonRoot() bool {
+	for i := range p.Pod.Spec.Containers {
+		securityContext := p.Pod.Spec.Containers[i].SecurityContext
+		if securityContext == nil || securityContext.RunAsNonRoot == nil {
+			continue
+		}
+		if *securityContext.RunAsNonRoot {
+			return true
+		}
+	}
+	return false
```

Comment:
I'm not sure if this is ok. As per k8s docs:

![image](https://github.com/test-network-function/cnf-certification-test/assets/87083379/73cead6b-8fe3-4bd7-be16-84ebea430418)

So if I'm not wrong, this means that some fields can be set in pod's SecurityContext and/or container's SecurityContext, where container's ones take precedence. So if "RunAsNonRoot" was not found in the container, we should also check whether that field exists in pod's SecurityContext.

Reply:
Yes good call.  What we were seeing in the hazelcast operator was set on the container but yes we should check both properties at the pod-level and the container-level.

Reply:
good point! i will add it.